web/modules/custom/dpl_opening_hours/src/Controller/OpeningHoursEditorController.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+namespace Drupal\dpl_opening_hours\Controller;
+
+use Drupal\Core\Access\AccessResult;
+use Drupal\Core\Controller\ControllerBase;
+use Drupal\node\NodeInterface;
+use Drupal\taxonomy\TermInterface;
+
+/**
+ * Defines OpeningHoursEditorController class.
+ */
+class OpeningHoursEditorController extends ControllerBase {
+
+  /**
+   * Display the opening hours app.
+   *
+   * @param int $node
+   *   The node ID.
+   *
+   * @return mixed[]
+   *   The app render array.
+   */
+  public function content(int $node): array {
+    // Load the node entity.
+    $nodeStorage = $this->entityTypeManager()->getStorage('node');
+    $nodeEntity = $nodeStorage->load($node);
+
+    if (!$nodeEntity instanceof NodeInterface || $nodeEntity->getType() !== 'branch') {
+      return [];
+    }
+
+    // Load taxonomy terms for opening_hours_categories.
+    $taxonomyStorage = $this->entityTypeManager()->getStorage('taxonomy_term');
+    $openingHoursCategories = $taxonomyStorage->loadByProperties([
+      "vid" => "opening_hours_categories",
+    ]);
+
+    if (!is_iterable($openingHoursCategories)) {
+      return [];
+    }
+
+    return [
+      '#theme' => 'dpl_react_app',
+      '#name' => 'opening-hours-editor',
+      '#data' => [
+        'opening-hours-editor-categories-config' => $this->buildOpeningHoursCategoriesArray($openingHoursCategories),
+        'opening-hours-branch-id-config' => $nodeEntity->id(),
+        'opening-hours-remove-event-button-text' => $this->t('Remove event'),
+      ],
+    ];
+  }
+
+  /**
+   * Build the opening hours categories array.
+   *
+   * @param \Drupal\taxonomy\TermInterface[] $openingHoursCategories
+   *   Array of opening hours categories.
+   *
+   * @return mixed[]
+   *   Array of category details.
+   */
+  private function buildOpeningHoursCategoriesArray($openingHoursCategories): array {
```

Comment:
```suggestion
  private function buildOpeningHoursCategoriesArray(array $openingHoursCategories): array {
```

The type specified in the DocBlock is nice as it includes the type. Having a proper type for the argument is a nice baseline.